pandas/plotting/_matplotlib/core.py

```diff
@@ -1345,6 +1348,16 @@ def _make_plot(self):
 
         for i, (label, y) in enumerate(self._iter_data(fillna=0)):
             ax = self._get_ax(i)
+
+            if self.orientation == "vertical":
```

Comment:
is there any assert in orientation that it takes on only certain values?

Reply:
No assert. Orientation is never a parameter. It is defined by the BarPlot class as "vertical", and overloaded by the Barhplot class as "horizontal". It is also defined in LinePlot class.
Globaly, it is use by the _post_plot_logic_common method from the base class MPLPlot.